### ISSUE TITLE
pgf backend doesn't set color of text when the color is black

### ISSUE THREAD

**Issue by stilley2** [matplotlib/matplotlib#11756]
### Bug report

**Bug summary**

When rendering black text with the pgf backed, the output is actually the default color set in the latex document. This issue comes up in beamer slides, where the default text color is not necessarily black.

**Code for reproduction**

```python
from matplotlib import figure
from matplotlib.backends.backend_pgf import FigureCanvasPgf


if __name__ == '__main__':
    fig = figure.Figure(figsize=(3, 3))
    ax = fig.add_subplot(1, 1, 1)
    ax.text(0.5, 0.3, 'test', color=(0, 0, 0))
    ax.text(0.5, 0.7, 'test', color=(0.01, 0.01, 0.01))
    ax.set_facecolor('blue')
    canvas = FigureCanvasPgf(fig)
    canvas.print_figure('test.pgf')
```

```latex
\documentclass{beamer}
\setbeamercolor{background canvas}{bg=black}
\setbeamercolor{normal text}{fg=white}
\usepackage{pgf}

\begin{document}
\begin{frame}{test frame}

dummy text

\input{test.pgf}

\end{frame}
\end{document}
```

**Actual outcome**

![cropped_screenshot](https://user-images.githubusercontent.com/17338580/43097815-40b58bfc-8e8b-11e8-9b14-bac0ba7fbeee.png)

**Expected outcome**

The white "test" should be black

**Matplotlib version**

  * Operating system: CentOS 7
  * Matplotlib version: 2.0.2 (conda-forge)
  * Matplotlib backend (`print(matplotlib.get_backend())`): pgf
  * Python version: 3.6.1
  * Jupyter version (if applicable):
  * Other libraries: 

**Possible solution** 
I think the issue is [here](https://github.com/matplotlib/matplotlib/blob/184a408fb45fdd02991ed517aa179113e3ade2b4/lib/matplotlib/backends/backend_pgf.py#L676) and that the `if` line should be removed

**Comment by anntzer** [comment]
Sounds reasonable, do you want to open a PR?

**Comment by stilley2** [comment]
I believe #11757 fixes the issue.

**Comment by dstansby** [comment]
Fixed by #11757
